src/libstd/collections/hash/map.rs

```diff
@@ -1818,6 +1818,100 @@ impl<'a, K, V> fmt::Debug for Drain<'a, K, V>
     }
 }
 
+/// A place for insertion to a `Entry`.
+///
+/// See [`HashMap::entry`](struct.HashMap.html#method.entry) for details.
+#[must_use = "places do nothing unless written to with `<-` syntax"]
+#[unstable(feature = "collection_placement",
+           reason = "struct name and placement protocol is subject to change",
+           issue = "30172")]
+pub struct EntryPlace<'a, K: 'a, V: 'a> {
+    bucket: Option<FullBucketMut<'a, K, V>>,
```

Comment:
I'm just lazy that I want to use existing `FullBucket::take` to remove the entry. It takes a `self` parameter. But in the `drop` method, there is only `&mut self`, the bucket field can't move.

It is fixed by adding another `FullBucket::remove` method, which takes a `&mut self` parameter. In `drop` method, I can call this `remove` now.